http://sourceware.org/ml/gdb-patches/2016-07/msg00179.html
Subject: [testsuite patch] Fix gdb.gdb/selftest.exp for C++-O2-g-built GDB


--ZGiS0Q5IWpPtfppv
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

Hi,

tested on Fedora 24 x86_64 after:
	./configure; make
That is: CFLAGS='-g -O2' CXXFLAGS='-g -O2'

OK for check-in?


Jan

--ZGiS0Q5IWpPtfppv
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline; filename=1

gdb/testsuite/ChangeLog
2016-07-16  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* gdb.gdb/selftest.exp (do_steps_and_nexts): Add "next over TRY" and
	"step into captured_main (args)".
	(test_with_self): Add "captured_main (args);" case.

diff --git a/gdb/testsuite/gdb.gdb/selftest.exp b/gdb/testsuite/gdb.gdb/selftest.exp
index af0026c..2cdd5c1 100644
--- a/gdb/testsuite/gdb.gdb/selftest.exp
+++ b/gdb/testsuite/gdb.gdb/selftest.exp
@@ -116,6 +116,14 @@ proc do_steps_and_nexts {} {
 		set description "next over lim_at_start initialization"
 		set command "next"
 	    }
+	    -re ".*TRY.*$gdb_prompt $" {
+		set description "next over TRY"
+		set command "next"
+	    }
+	    -re ".*captured_main \\(args\\);.*$gdb_prompt $" {
+		set description "step into captured_main (args)"
+		set command "step"
+	    }
 	    -re ".*count . 0x3.*$gdb_prompt $" {
 		set description "next over conditional stack alignment code 1"
 		set command "next"
@@ -330,6 +338,13 @@ proc test_with_self { executable } {
 	-re "Starting program.*Breakpoint \[0-9\]+,.*captured_main .data.*$gdb_prompt $" {
 	    xfail "$description (line numbers scrambled?)"
 	}
+	-re "captured_main \\(args\\);\r\n$gdb_prompt $" {
+	    gdb_test_multiple "step" "$description" {
+		-re "captured_main .data.* at .*main.c:.*$gdb_prompt $" {
+		    pass "$description"
+		}
+	    }
+	}
 	-re "vfork: No more processes.*$gdb_prompt $" {
 	    fail "$description (out of virtual memory)"
 	    set timeout $oldtimeout

--ZGiS0Q5IWpPtfppv--

